packages/framework/src/Resources/scripts/common/validation/customizeBundle.js

```diff
@@ -115,6 +115,22 @@
         Shopsys.timeout.setTimeoutAndClearPrevious('Shopsys.validation.validateWithParentsDelayed', executeDelayedValidators, 100);
     };
 
+    // Issue in dynamic collections validation
+    // https://github.com/formapro/JsFormValidatorBundle/issues/139
+    FpJsFormValidator._preparePrototype = FpJsFormValidator.preparePrototype;
+    FpJsFormValidator.preparePrototype = function (prototype, name, id) {
+        prototype.name = prototype.name.replace(/__name__/g, name);
```

Comment:
Could you please add some explanation what exactly happens here, i.e. what was the problem and how it was fixed? I see that you replace both prototype.name and prototype.id with given name but I do not understand why :slightly_frowning_face: